### PR TITLE
Fixed-precision floats

### PR DESCRIPTION
**Pull request by fredrik-johansson**
Work in progress. Sample timings and speedup vs ``arf``:

```
$ build/nfloat/profile/p-vs_arf 
                   _gr_vec_sum          _gr_vec_product      _gr_vec_dot
prec = 64
n =    3        1.080e-08 (2.352x)   8.040e-09 (3.483x)   1.330e-08 (1.880x)
n =   10        4.550e-08 (2.374x)   3.300e-08 (3.576x)   2.420e-08 (1.967x)
n =  100        3.990e-07 (3.208x)   4.070e-07 (3.317x)   1.870e-07 (1.898x)
prec = 128
n =    3        2.320e-08 (1.815x)   9.510e-09 (3.270x)   1.650e-08 (1.988x)
n =   10        8.970e-08 (1.973x)   3.860e-08 (3.420x)   3.780e-08 (1.725x)
n =  100        9.430e-07 (2.312x)   5.130e-07 (2.943x)   3.080e-07 (1.623x)
prec = 256
n =    3        2.410e-08 (2.178x)   1.720e-08 (2.244x)   4.890e-08 (1.509x)
n =   10        9.900e-08 (2.222x)   7.110e-08 (2.363x)   1.350e-07 (1.474x)
n =  100        9.740e-07 (2.813x)   8.650e-07 (2.335x)   1.270e-06 (1.433x)
prec = 512
n =    3        2.940e-08 (2.082x)   3.280e-08 (2.098x)   7.870e-08 (1.576x)
n =   10        1.090e-07 (2.257x)   1.420e-07 (2.106x)   2.350e-07 (1.570x)
n =  100        1.160e-06 (2.534x)   1.600e-06 (2.163x)   2.270e-06 (1.559x)
prec = 1024
n =    3        3.460e-08 (1.934x)   1.500e-07 (1.393x)   2.440e-07 (1.488x)
n =   10        1.370e-07 (1.978x)   7.130e-07 (1.289x)   7.990e-07 (1.389x)
n =  100        1.450e-06 (2.324x)   8.010e-06 (1.273x)   7.900e-06 (1.392x)
prec = 2048
n =    3        5.130e-08 (1.474x)   4.720e-07 (1.419x)   7.510e-07 (1.232x)
n =   10        1.860e-07 (1.790x)   2.170e-06 (1.373x)   2.450e-06 (1.233x)
n =  100        1.930e-06 (2.010x)   2.420e-05 (1.360x)   2.450e-05 (1.229x)
```

### CODE REVIEW

File: doc/source/nfloat.rst [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+          nfloat1024_struct
+          nfloat2048_struct
+          nfloat64_t
+          nfloat128_t
+          nfloat192_t
+          nfloat256_t
+          nfloat384_t
+          nfloat512_t
+          nfloat1024_t
+          nfloat2048_t
```
Comment (albinahlback) [inline doc/source/nfloat.rst:86]: I suppose the numbers will be half of this for 32-bit systems? Or is this module only available for 64-bit computers?

Comment (albinahlback) [reply doc/source/nfloat.rst:86]: My bad. I suppose this just assumes 64-bit system.

File: doc/source/nfloat.rst [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    | exponent limb |
+    +---------------+
+    |   sign limb   |
+    +---------------+
+    |  mantissa[0]  |
+    +---------------+
+    |      ...      |
+    +---------------+
+    | mantissa[n-1] |
+    +---------------+
```
Comment (albinahlback) [inline doc/source/nfloat.rst:22]: Looking at `nfloat_one`, I understand it as they are normalized in the same manner as the arf-objects (apart from removing trailing zero limbs). Could this perhaps be stated here?

File: doc/source/nfloat.rst (hunk 1 above)
Comment (fredrik-johansson) [reply doc/source/nfloat.rst:86]: No, those types should work the same on both 32-bit and 64-bit systems (except for minor rounding differences). They just have twice as many limbs on 32-bit.

File: doc/source/nfloat.rst (hunk 2 above)
Comment (fredrik-johansson) [reply doc/source/nfloat.rst:22]: Yes, good point, that ought to be documented explicitly.

**Comment by fredrik-johansson** [comment]
There's more to do here, but this should be good enough for an initial merge.

**Pull request merged**